src/seq/index.rs

```diff
@@ -249,6 +252,114 @@ where R: Rng + ?Sized {
     }
 }
 
+/// Randomly sample exactly `amount` distinct indices from `0..length`, and
+/// return them in an arbitrary order (there is no guarantee of shuffling or
+/// ordering). The weights are to be provided by the input function `weights`,
+/// which will be called once for each index.
+///
+/// This method is used internally by the slice sampling methods, but it can
+/// sometimes be useful to have the indices themselves so this is provided as
+/// an alternative.
+///
+/// This implementation uses `O(length)` space and `O(length)` time if the
```

Comment:
More precisely, it's `O(length + amount)` space, right?

Reply:
Technically since we know that `amount <= length`, I think `O(length)` is technically correct, but I agree that `O(length + amount)` is more useful for readers.